src/erlazure_insights.erl

```diff
@@ -72,17 +72,28 @@ request_api(Url, Config) ->
 http_request(SubscriptionId, Filter, Select, Config) ->
     URL = "https://management.azure.com/subscriptions/" ++
     SubscriptionId ++
-    "/providers/microsoft.insights/eventtypes/management/values?" ++ "api-version=2014-04-01" ++ 
-    "&%24filter=" ++ Filter ++
-    "&%24select=" ++ Select,
+    "/providers/microsoft.insights/eventtypes/management/values?" ++
+    make_query_string([
+            {"api-version", "2014-04-01"},
+            {"$filter", Filter},
+            {"$select", Select}
+        ]),
     request_api(URL, Config).
 
+make_query_string(Params) ->
+    lists:foldl(fun({Key, Value}, Acc) -> 
+                case Value of
+                    [] -> Acc;
+                    _ -> Acc ++ "&" ++ Key ++ "=" ++ http_uri:encode(Value)
```

Comment:
also I'm concerned about the fact we always concatenate things on the right:

```
Acc ++ Something
```

which is expensive, since we need to copy `Acc` on each iteration - look there http://erlang.org/doc/efficiency_guide/listHandling.html